src/arithmetic/entity_funcs/entity_funcs.c

```diff
@@ -140,43 +140,67 @@ SIValue AR_EXISTS(SIValue *argv, int argc, void *private_data) {
 	return SI_BoolVal(1);
 }
 
-SIValue _AR_NodeDegree(SIValue *argv, int argc, GRAPH_EDGE_DIR dir) {
-	if(SI_TYPE(argv[0]) == T_NULL) return SI_NullVal();
-	Node *n = (Node *)argv[0].ptrval;
-	Edge *edges = array_new(Edge, 0);
-	GraphContext *gc = QueryCtx_GetGraphCtx();
+// returns node incoming/outgoing degree
+static SIValue _AR_NodeDegree
+(
+	SIValue *argv,
+	int argc,
+	GRAPH_EDGE_DIR dir  // edge direction
+) {
+	ASSERT(SI_TYPE(argv[0]) != T_NULL);
+
+	Node          *n     = (Node*)argv[0].ptrval;
+	uint64_t      count  = 0;
+	GraphContext  *gc    = QueryCtx_GetGraphCtx();
 
 	if(argc > 1) {
-		// We're interested in specific relationship type(s).
+		// we're interested in specific relationship type(s)
 		for(int i = 1; i < argc; i++) {
 			const char *label = argv[i].stringval;
 
-			// Make sure relationship exists.
+			// make sure relationship exists.
 			Schema *s = GraphContext_GetSchema(gc, label, SCHEMA_EDGE);
-			if(!s) continue;
+			if(s == NULL) {
+				continue;
+			}
 
-			// Accumulate edges.
-			Graph_GetNodeEdges(gc->g, n, dir, s->id, &edges);
+			// count edges
+			count += Graph_GetNodeDegree(gc->g, n, dir, s->id);
```

Comment:
I can't find the definition for this function for some reason. Did you forget to add/commit/push it? :-)

Reply:
@vityafx See graph.c

Reply:
I don't have it, perhaps, I am on an old master.

Reply:
Oh, weird! It wasn't showing up here for some reason. I checked a few times! I can see it here now in this PR.